### PR TITLE
Start adding zero-copy docs

### PR DESCRIPTION
**Pull request by CraftSpider**
Adds documentation to all zero_copy items ~~And one or two minor changes I couldn't resist while documenting~~

### CODE REVIEW

File: src/zero_copy/primitive.rs [hunk 1]
```diff
@@ -1,7 +1,57 @@
...
+///
+/// This parser is very useful when you wish to force a parser to consume *all* of the input. It is typically combined
+/// with [`Parser::then_ignore`].
+///
+/// The output type of this parser is `()`.
+///
+/// # Examples
+///
+/// ```
+/// # use chumsky::prelude::*;
```
Comment (bew) [inline src/zero_copy/primitive.rs:31]: hmm shouldn't you be using the zero copy prelude instead?
```suggestion
/// # use chumsky::zero_copy::prelude::*;
```
(same for ALL examples)

Comment (CraftSpider) [reply src/zero_copy/primitive.rs:31]: Correct, I haven't yet updated most of the examples. I'm currently working on documenting things new in zero_copy, then once that's done, I plan to do a massive search-and-replace before cleaning up all the errors that causes

**Comment by CraftSpider** [comment]
Unrelated note: After this PR, definitely think it might be a good idea to pull `Container`, `Seq`, etc into their own module, to track these common 'container helpers'.

**Comment by CraftSpider** [comment]
Currently `recover_with` fails, and two docs are TODO. On the other hand, I think it may be worth to merge now and fix those things later, since this change is very prone to bitrot

**Comment by zesterer** [comment]
Do you think it's ready for merge?

**Comment by CraftSpider** [comment]
I'm going to add at least basic documentation to the two TODO ones quick, then I think so

**Comment by CraftSpider** [comment]
I'd call this PR ready to merge - there are one or two rough spots, but overall I think this improves the current state of zero-copy documentation.

**Comment by zesterer** [comment]
Thanks! This is a big step forward :)

**Pull request merged**